crates/configuration/src/schema/mod.rs

```diff
@@ -69,13 +69,15 @@ impl From<Type> for ndc_models::Type {
                 // ExtendedJSON can respresent any BSON value, including null, so it is always nullable
                 Type::ExtendedJSON => ndc_models::Type::Nullable {
                     underlying_type: Box::new(ndc_models::Type::Named {
-                        name: mongodb_support::EXTENDED_JSON_TYPE_NAME.to_owned(),
+                        name: mongodb_support::EXTENDED_JSON_TYPE_NAME.to_owned().into(),
```

Comment:
I think for cases like this we might avoid heap allocation with something like,

```rust
TypeName(SmolStr::new_static(mongodb_support::EXTENDED_JSON_TYPE_NAME))
```

That would be a bit easier if the ndc-models `newtype` macro added implementations for `From SmolStr` or `From &'static str`. But it looks like the ref-cast crate helps to fill that gap?

This is absolutely not worth you spending time on making changes. I'm commenting only because I enjoy the allocation golf thought exercises.